### ISSUE TITLE
Optional types have to implement default

### ISSUE THREAD

**Issue by ThomasAlban** [elastio/bon#12]
If I do this:
```rust
struct TestType;

#[builder]
fn test(test_type: Option<TestType>) {}

fn run() {
    test().call();
}
```
I get this error:
```
the method `call` exists for struct `TestBuilder`, but its trait bounds were not satisfied
the following trait bounds were not satisfied:
`TestType: std::default::Default`
which is required by `bon::private::Optional<viewer::kmp::checkpoints::TestType>: bon::private::IntoSet<std::option::Option<viewer::kmp::checkpoints::TestType>>`
```
Am I missing something? Why do optional types need to implement `Default`?

**Comment by ThomasAlban** [comment]
Interestingly, if you change it to this:
```rust
struct TestType;

#[builder]
fn test(test_type: Option<TestType>) {}

fn run() {
    test().maybe_test_type(None::<TestType>).call();
}
```
the error goes away. If you just do `None` and not `None::<TestType>` there is an error which is strange!

**Comment by Veetaha** [comment]
Thank you for the bug report! It's a really small bug in `bon`, I'll fix it shortly and make a patch release, thank you!

**Comment by Veetaha** [comment]
The fix has been released in a new patch version `1.0.2`.
Feel free to reopen the issue if the problem persists.

> If you just do `None` and not `None::<TestType>` there is an error which is strange!

This is a bit different problem that is inherent to the setter method signature:
```rust
fn maybe_test_type(self, value: Option<impl Into<TestType>>) -> NextBuilderState
```

Because it accepts an `Option<impl Into<TestType>>` the type of `None` needs to explicitly specified unless inferred by the compiler from other context.
